src/BloomBrowserUI/bookEdit/OverflowChecker/OverflowChecker.ts

```diff
@@ -0,0 +1,217 @@
+        // In fact, the focused grey border causes the same problem in detecting the bottom of a marginBox
+        // so we'll apply the same 'fudge' factor to both comparisons.
+        var focusedBorderFudgeFactor = 2;
+
+        return  elemBottom > parentBottom + focusedBorderFudgeFactor;
+    }
+
+    // Checks for overflow on a bloom-page and adds/removes the proper class
+    // N.B. This function is specifically designed to be called from within AddOverflowHandler()
+    // but is also called from within StyleEditor (and therefore public)
+    public static MarkOverflowInternal(box) {
+        // There are two types of overflow that we need to check.
+        // 1-When we're called by a handler on an element, we need to check that that element
+        // doesn't overflow internally (i.e. has too much stuff to fit in itself).
+        // 2-We also need to check that this element and any OTHER elements on the page
+        // haven't been pushed outside the margins
+
+        // Type 1 Overflow
+        var $box = $(box);
+        $box.removeClass('overflow');
+        if (OverflowChecker.IsOverflowingSelf(box)) {
+            $box.addClass('overflow');
+        } else {
+            //now, thing is, while the text may fit in our box, our box may not fit our parent. Or grandparent, etc.
+            //It could be that we could just do this on up the hierarchy? For now, here's the case we know is important,
+            // in the Origami pages, where the space allocated could be too small.
+            if (OverflowChecker.HasSplitParentThatOverflows($box)) {
+                $box.addClass('overflow');
+            }
+        }
+
+        var container = $box.closest('.marginBox');
+        //NB: for some historical reason in March 2014 the calendar still uses textareas
+        var queryElementsThatCanOverflow = ".bloom-editable, textarea";
+        var editablePageElements = $(container).find(queryElementsThatCanOverflow);
+
+        // Type 2 Overflow - We'll check ALL of these for overflow past marginBox
+        editablePageElements.each(function() {
+            var $this = $(this);
+            if (OverflowChecker.IsOverflowingMargins($this[0])) {
+                $box.addClass('overflow'); // probably typing in the focused element caused this
+                $this.addClass('overflow'); // but it's this one that is actually overflowing
+            }
+            else {
+                if (!OverflowChecker.IsOverflowingSelf($this[0]) && !OverflowChecker.HasSplitParentThatOverflows($this)) {
+                    $this.removeClass('overflow'); // might be a remnant from earlier overflow
+                }
+            }
+        });
+        OverflowChecker.UpdatePageOverflow(container.closest('.bloom-page'));
+    } // end MarkOverflowInternal
+
+    static HasSplitParentThatOverflows(jQueryBox : JQuery) : boolean {
+        var splitterParents = jQueryBox.parents('.split-pane-component-inner');
```

Comment:
The name of this method suggests it is checking all parents, but it actually only checks the first. ParentIsSplitThatOverflows? might better suggest that it's only looking at the most immediate parent.
Is that all you need to consider?